proto/regen/ecocredit/v1/query.proto

```diff
@@ -72,35 +102,56 @@ service Query {
   // BatchesByIssuer queries all batches issued from a given issuer address.
   rpc BatchesByIssuer(QueryBatchesByIssuerRequest)
       returns (QueryBatchesByIssuerResponse) {
-    option (google.api.http).get =
-        "/regen/ecocredit/v1/batches/issuer/{issuer}";
+    option (google.api.http) = {
+      get : "/regen/ecocredit/v1/batches-by-issuer/{issuer}"
+      additional_bindings : [
+        {get : "/regen/ecocredit/v1/batches/issuer/{issuer}"}
+      ]
+    };
   }
 
   // BatchesByClass queries all batches issued from a given class.
   rpc BatchesByClass(QueryBatchesByClassRequest)
       returns (QueryBatchesByClassResponse) {
-    option (google.api.http).get =
-        "/regen/ecocredit/v1/batches/class/{class_id}";
+    option (google.api.http) = {
+      get : "/regen/ecocredit/v1/batches-by-class/{class_id}"
+      additional_bindings : [
+        {get : "/regen/ecocredit/v1/batches/class/{class_id}"}
+      ]
+    };
   }
 
   // BatchesByProject queries for all batches from a given project with
   // pagination.
   rpc BatchesByProject(QueryBatchesByProjectRequest)
       returns (QueryBatchesByProjectResponse) {
-    option (google.api.http).get =
-        "/regen/ecocredit/v1/batches/project/{project_id}";
+    option (google.api.http) = {
+      get : "/regen/ecocredit/v1/batches-by-project/{project_id}"
+      additional_bindings : [
+        {get : "/regen/ecocredit/v1/batches/project/{project_id}"}
+      ]
+    };
   }
 
   // Batch queries for information on a credit batch.
   rpc Batch(QueryBatchRequest) returns (QueryBatchResponse) {
-    option (google.api.http).get = "/regen/ecocredit/v1/batches/{batch_denom}";
+    option (google.api.http) = {
+      get : "/regen/ecocredit/v1/batch/{batch_denom}"
+      additional_bindings : [
+        {get : "/regen/ecocredit/v1/batches/{batch_denom}"}
+      ]
+    };
   }
 
   // Balance queries the balance (both tradable and retired) of a given credit
   // batch for a given account address.
   rpc Balance(QueryBalanceRequest) returns (QueryBalanceResponse) {
-    option (google.api.http).get =
-        "/regen/ecocredit/v1/batches/{batch_denom}/balance/{address}";
+    option (google.api.http) = {
+      get : "/regen/ecocredit/v1/balance/{batch_denom}/{address}"
```

Comment:
below we have a `Balances` query of the form `"/regen/ecocredit/v1/balances/{address}"`, I think we should be consistent and change this to `"/regen/ecocredit/v1/balances/{address}/{batch_denom}"`

Reply:
This is for all balances. We don't specify a batch denom here.

Reply:
The below is for all balance. We don't specify a batch denom when querying all balances.

Reply:
The below is for all balances. We don't specify a batch denom when querying all balances.

Reply:
Ah, I see. Although I think we should be consistent with using the query name for the first `get`. Maybe we should add another binding to support what you're suggesting.

Reply:
https://github.com/regen-network/regen-ledger/pull/1168/commits/3131ea9da5ae591f039979970dce91709cbc3eab